tests/compile-fail/data_race/relax_acquire_race.rs

```diff
@@ -0,0 +1,43 @@
+// ignore-windows: Concurrency on Windows is not supported yet.
+
+use std::thread::spawn;
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+#[derive(Copy, Clone)]
+struct EvilSend<T>(pub T);
+
+unsafe impl<T> Send for EvilSend<T> {}
+unsafe impl<T> Sync for EvilSend<T> {}
+
+static SYNC: AtomicUsize = AtomicUsize::new(0);
+
+pub fn main() {
+    let mut a = 0u32;
+    let b = &mut a as *mut u32;
+    let c = EvilSend(b);
+
+    unsafe {
+        let j1 = spawn(move || {
+            *c.0 = 1;
+            SYNC.store(1, Ordering::Release);
+        });
+
+        let j2 = spawn(move || {
+            if SYNC.load(Ordering::Acquire) == 1 {
+                SYNC.store(2, Ordering::Relaxed);
+            }
+        });
+
+        let j3 = spawn(move || {
+            if SYNC.load(Ordering::Acquire) == 2 {
+                *c.0 //~ ERROR Data race
```

Comment:
Isn't this test very scheduler-dependent? And to make them robust the threads would have to `loop` if they `SYNC.load` the wrong thing?

If so, no need to change the code, but please leave a comment so that if/when we change the scheduler, we have an idea for what to do here.